libraries/DataFlash/DataFlash_MAVLink.cpp

```diff
@@ -500,16 +509,21 @@ void DataFlash_MAVLink::do_resends(uint32_t now)
     }
     uint32_t oldest = now - 100; // 100 milliseconds before resend.  Hmm.
     while (count_to_send-- > 0) {
+        if (!semaphore->take_nonblocking()) {
```

Comment:
Yeah, it was intentional.  I wanted to give any process which wanted to do something useful (log a message) a chance to do something in preference to something doing this relatively useless, retryable background task.